### ISSUE TITLE
DSA spec: Add domainSeed to examples for PQGen and PQVer with Provable Primes

### ISSUE THREAD

**Issue by smuellerDD** [usnistgov/ACVP#359]
For PQ gen with provable primes, I return the following:

```
      {
        "tcId":1,
        "p":"b8055471e1cbdd295329dac5c1d568d0f63a6e0af89382b1a423d99091134554ced58bbaacbc9939d54f63c42f3a5aba5632f5f6dec7e657c8831d3e2c502052e8afed94af7234352cb05f680944fc162ca9468fb193ddfd423b2c8fd373b3dda652dd6280fc3a52845c6552576336eb1b059e0afc5526fa95d04c48414a270e5aea3bfe3efd6ed58f944dc5a25b3f2f8e1b95b078c4fd3cf05f1370d100c402d60c3da455b4c1faa78aa7ec3dfd5305e9c03d5b6caacf8e245c0cf561b5b9523499ff4c42b555165e0efbb4529f44c285be08acd623bc6cd5cb21bbd0e4ee063c75e206ed3f8c80a2bb95e6e5fc22bb7bfefc1f78b5e634ba1b4f8e64590073",
        "q":"c12d56572acb67d6d2cecbc64295b9dffba7238bfa331d367389da9d",
        "pSeed":"4a7f676a4c2e44a17eaa4eb34ad5db27fd3607c224862ee1d970f16ea8ca1e7a482a2f5d16a982fd8681076fd831421dd789b774032f016b96a23ad05245dd0d51d2299ef15587fe19cb5e2fdbaee4920d703c17821add3b2bd0683e8cada46383b9d17758422e594d15a4a21fa85fb95ce4da58603b2d1af9828661a4092496766528f637b562569e3d0c026128c3b04115a47c03384319f5d4ee160a1efa1e3fa1f3e40126af38ac251854a375bf470da86ebd281997f1f68b3d80c62618477b08aa34ec00c242f53524ec224a48a1de5650035ee6304d6942cdb115b88861",
        "qSeed":"4a7f676a4c2e44a17eaa4eb34ad5db27fd3607c224862ee1d970f16ea8ca1e7a482a2f5d16a982fd8681076fd831421dd789b774032f016b96a23ad05245dd0d51d2299ef15587fe19cb5e2fdbaee4920d703c17821add3b2bd0683e8cada46383b9d17758422e594d15a4a21fa85fb95ce4da58603b2d1af9828661a4092496766528f637b562569e3d0c026128c3b04115a47c03384319f5d4ee160a1efa1e3fa1f3e40126af38ac251854a375bf470da86ebd281997f1f68b3d80c62618477b08aa34ec00c242f53524ec224a48a1de5650035ee6304d6942cdb115b88112",
        "pCounter":681,
        "qCounter":188
      },
```

This seems to be consistent with the DSA spec.

Yet, the server responds as verdict with the following:

```
      "tcId" : 1,
      "result" : "fail",
      "reason" : "Unexpected failure"
```

vsIds: 2600, 2605, 2610

**Comment by smuellerDD** [comment]
This seems to be an issue with provable primes, too. My answer:

```
      {
        "tcId":1,
        "p":"b5f670127d7adf6163134aab9c1c036ee5d321339f73fba3443c2f3a0768b8d24e85c683837484eb93ea5cfa32f1e3dae88025221b2cfd2962782a2b69c721a1a31280e6173105dc14abdc33f58815de5a4bcf70da56d94533823b30fea4e728e434bd14372488368e0a6681b5f77690c95b96e0c8602ac0411b85e7b5631e9b7be31b80491075d188a6bfa1e7d1b6a8d2699ecfaee435773f8469cd08cd8aa632e38db02ba9e200c63310a1fe58d9704efa69e65780a8833eeb1f4efd7a876a25bbf254b705c19758c8f3393f57bc86e01da00d51e27e476b33a543d67894f027ab6136a88842ede48892c27363dcb3f847bad225adcc4c0f060a92e18e62c3",
        "q":"b785bbd8b5ea35eb2cc32b4d775b412c784f4413b7b78c6216643c51",
        "counter":177
      },
      {
        "tcId":2,
        "p":"9dddb75a56770d4ab79baa79e0e267511fc4cadd79814c2fe4a10c28f155ff5cd5ba3b9a4a5a5b695a4887a0322d9a2a3a45bd645e510bb976fb9ca222814e9d1cad36a143f9ea8cd8fbff9dabedbb621b9a17a1a06b66bc9a883c67e22e075dcb699e7ca56ea292988b50148d6ea6ee570bd85f9a077253cbb8302c3edee5a2addd880979bb081dab45fb2b7091ccbed8a83b3cd134b1d0e75d966ebf9402bd8408db25f43e86749430a512c42d4aa80a5b4f667aec468477fd5b903d3e91718ebb4b94daaacacc6e905a7c19937ca86b55f9e0c3d596456167f9138951f0f5725193f0dcfab54f1360bd5b77337a24688d8af5ad5b72e438006cfc2a5c01ff",
        "q":"81da3b0c33904f1acddeab899daedcbf3c8d4dfedbc404b77bcc34df",
        "counter":24
      },
```

The server answers:

```
      "tcId" : 1,
      "result" : "fail",
      "reason" : "Unexpected failure"
    }, {
      "tcId" : 2,
      "result" : "fail",
      "reason" : "Unexpected failure"
```

vsID 2548

**Comment by celic** [comment]
This is happening because the server is expecting the results for `tcId` 1 and 2 to contain a `domainSeed` value in addition to the `pSeed` and `qSeed` you provide in these vsIds.

**Comment by celic** [comment]
This is a gen/val error. I'll update when a fix gets deployed.

**Comment by celic** [comment]
<previous message deleted>

Sorry to go back and forth here. `domainSeed` is necessary for validation of PQGen using the `provable` method. This is in addition to `pSeed` and `qSeed`. I don't believe there is an error on the gen/val side. There could be a spec issue though.

**Comment by smuellerDD** [comment]
The example in A.6 does not list the domainSeed for tcId 1 and 2.

I will add it in my code base and report back.

**Comment by smuellerDD** [comment]
confirmed: after providing the dmainSeed, the verification passes.

**Comment by celic** [comment]
Thanks for the confirmation. I've listed this as a spec error only.

**Comment by smuellerDD** [comment]
Fixed with pull request #417.
